cpp/tests/filling/filling_tests.cu

```diff
@@ -35,13 +35,16 @@ TYPED_TEST_CASE(FillingTest, test_types);
 
 constexpr gdf_size_type column_size{1000};
 
-template <typename T>
+auto all_valid = [](gdf_index_type row) { return true; };
+
+template <typename T, typename BitInitializerType = decltype(all_valid)>
```

Comment:
I don't think constraining the bit initializer to be exactly the same type as `all_valid` is necessary. Just leave it unconstrained -- if you pass something that doesn't work it won't compile.

```suggestion
template <typename T, typename BitInitializerType>
```

Reply:
@harrism, I originally had it unconstrained, but the compiler had a hard time with it: 

```
error: no instance of function template "FillTest" matches the argument list
            argument types are: (gdf_index_type, int, int32_t, __nv_bool)
```

At this stage I think the issue is because of 2 different defaulted parameters and type deduction, because when I do something like this it works:

```
auto all_valid = [](gdf_index_type row) { return true; };

template <typename T>
void FillTest(gdf_index_type begin, gdf_index_type end,
              T value, bool value_is_valid = true)
{
  FillTest_(begin, end, value, value_is_valid, all_valid);
}

template <typename T, typename BitInitializerType>
void FillTest(gdf_index_type begin, gdf_index_type end,
              T value, bool value_is_valid ,
              BitInitializerType source_validity = all_valid)
{
  FillTest_(begin, end, value, value_is_valid, source_validity);
}

template <typename T, typename BitInitializerType>
void FillTest_(gdf_index_type begin, gdf_index_type end,
              T value, bool value_is_valid,
              BitInitializerType source_validity)
{
  column_wrapper<T> source(column_size,
    [](gdf_index_type row) { return static_cast<T>(row); },
    [&](gdf_index_type row) { return source_validity(row); });
...
```

Any suggestions?

Reply:
Just leave it. If I were doing it, I would have just made FillTest take a source column_wrapper as a parameter rather than passing a functor.